src/stellarmesh/moab.py

```diff
@@ -532,18 +567,61 @@ def from_mesh(
         core = pymoab.core.Core()
         model = cls(core)
 
-        known_surfaces: dict[int, DAGMCSurface] = {}
-
         with mesh:
             # Warn about volume elements being discarded
-            _, element_tags, _ = gmsh.model.mesh.get_elements(3)
-            if element_tags:
+            if gmsh.model.mesh.get_elements(3)[1]:
                 logger.warning("Discarding volume elements from mesh.")
 
+            # 1. Add surface elements and boundary conditions
+            surfaces: dict[int, DAGMCSurface] = {}
+            surface_dimtags = gmsh.model.get_entities(2)
+            surface_tags = [s[1] for s in surface_dimtags]
+            for i, surface_tag in enumerate(surface_tags):
+                surface_set = model.create_surface(surface_tag)
+                surfaces[surface_tag] = surface_set
+
+                surface_groups = gmsh.model.get_physical_groups_for_entity(
+                    2, surface_tag
+                )
+                if (num_groups := len(surface_groups)) not in [0, 1]:
+                    raise ValueError(
+                        f"Surface with tag {surface_tag} and global_id {i}"
+                        f"belongs to {num_groups} physical groups, should be 0 "
+                        f"or 1."
+                    )
+                elif num_groups == 1:
+                    boundary_name = gmsh.model.get_physical_name(2, surface_groups[0])
+                    surface_set.boundary = boundary_name[9:]
+
+                volume_adjacencies_dimtags = gmsh.model.get_adjacencies(2, surface_tag)[
+                    0
+                ]
+                if len(volume_adjacencies_dimtags) == 0:
+                    logger.warning(
```

Comment:
@paulromano Is this a sensible workaround. We cannot perform a DAGMC tally on a surface without an associated volume, but simply adding a "dummy" volume with "mat:void" works well. Is this acceptable?